### ISSUE TITLE
maybe a error in LabelSmoothing

### ISSUE THREAD

**Issue by huangnengCSU** [harvardnlp/annotated-transformer#18]
the forward() in class LabelSmoothing throws a bug when run the block [29], can you check the code of condition judgement "if mask.dim() > 0:" should be "if mask.nelement() > 0:" .because if the mask is tensor([]), then the dim() is equal 1, and number of element is equal 0.

the following  is my modified code, then there is no bug like before.
def forward(self, x, target):
        assert x.size(1) == self.size
        true_dist = x.data.clone()
        true_dist.fill_(self.smoothing / (self.size - 2))
        true_dist.scatter_(1, target.data.unsqueeze(1), self.confidence)
        true_dist[:, self.padding_idx] = 0
        mask = torch.nonzero(target.data == self.padding_idx)
        if mask.nelement() > 0:
            true_dist.index_fill_(0, mask.squeeze(), 0.0)
        self.true_dist = true_dist
        return self.criterion(x, Variable(true_dist, requires_grad=False))
